schemas/com.snowplowanalytics.snowplow/button_click/jsonschema/1-0-0

```diff
@@ -0,0 +1,60 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for a button click event",
+  "self": {
+    "vendor": "com.snowplowanalytics.snowplow",
+    "name": "button_click",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
+    "action": {
+      "type": "string",
+      "description": "The action associated with the button click",
```

Comment:
How would we know the action? Is it something that needs to be passed from the user or can be identified automatically?

Reply:
It would have to be identifier by the user. I was thinking initially that we could just take the text on the button as an "action", but there are enough icon buttons out there that that's probably not do-able, even with something like aria-labels, since they're often missing. 

I was thinking was that this could be added via a data attribute (something like `data-sp-action` on a button?), or with the usual `<button onClick="() => trackButtonClick({action: "foo"})"></button>`.

Reply:
If it's intended to be the text on the button, then maybe it's better to call it the `label`? But then I'm not sure how would this be different from the `elementContent`.

I think it would make sense if, by default, the text of the button was used here. That is the most common way to describe what a button does (on all platforms) so probably should be the only required property in the schema.

It's true that in some cases, buttons may not have text, but given that we limit the scope of this schema to `button_click` as opposed to `click`, that should be rare enough. And for those cases we could let users provide the label manually in some way.

Reply:
Sounds good, `action` has been renamed `label` and we can take the text on the button if present by default, and provide a way for the user to provide a label manually.